cedar-policy-core/src/evaluator/err.rs

```diff
@@ -30,10 +31,13 @@ pub struct EvaluationError {
     error_kind: EvaluationErrorKind,
     /// Optional advice on how to fix the error
     advice: Option<String>,
+    /// Source location of the error. (This overrides other sources if present,
```

Comment:
This seems error-prone to me. Maybe create an issue to fix it?

Reply:
Happy to create an issue, curious what you find to be error-prone about this?

Are you concerned that the source location in this field might disagree with the source location embedded in `error_kind`?  If it makes you feel better, the vast majority of the `EvaluationErrorKind`s do not have a source loc field -- the only cases that do have a source loc are `InvalidRestrictedExpression` and `NonValue`, and those just because they store an `Expr` as part of the error.  We could make an issue to remove these somehow, so that `error_kind` never contains a source loc; is that what you're suggesting?

Going the other way -- having all the source locs in `error_kind` so that we never need this field -- is backwards; we established recently that we prefer the convention to use the `*Kind` pattern to have a single place the source loc field is defined, rather than adding a source loc field in N places, once for each enum variant.  The struct here conforms to that practice.

Reply:
Yeah, it's just a general concern about "code smell," not anything concrete. Even if we always remember to keep them in sync, I'd worry about that complexity hiding other bugs. E.g., I think the bug we had where it was possible to add a link and a template with colliding names wouldn't have happened if we didn't have duplicate information in our lists of policies+templates and links+templates. Even though it wasn't caused by forgetting to keep data in sync, the added complexity (e.g., rolling back changes to one representation if modifying the other failed) distracts code reviewers from the obvious "you forgot to check that no link with this ID exists."